src/lib/model/InboundThirdpartyTransactionModel.js

```diff
@@ -0,0 +1,109 @@
+/**************************************************************************
+ *  (C) Copyright ModusBox Inc. 2019 - All rights reserved.               *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       Sridhar Voruganti - sridhar.voruganti@modusbox.com               *
+ **************************************************************************/
+
+'use strict';
+
+const {
+    BackendRequests,
+    HTTPResponseError,
+} = require('@internal/requests');
+
+const {
+    MojaloopRequests,
+    Ilp,
+    Errors,
+} = require('@mojaloop/sdk-standard-components');
+
+const shared = require('@internal/shared');
+
+/**
+ *  Models the operations required for inbound third party transaction requests
+ */
+class InboundThirdpartyTransactionModel {
+    constructor(config) {
+        this._cache = config.cache;
+        this._logger = config.logger;
+        this._dfspId = config.dfspId;
+        this._expirySeconds = config.expirySeconds;
+
+        this._mojaloopRequests = new MojaloopRequests({
+            logger: this._logger,
+            peerEndpoint: config.peerEndpoint,
+            alsEndpoint: config.alsEndpoint,
+            quotesEndpoint: config.quotesEndpoint,
+            transfersEndpoint: config.transfersEndpoint,
+            dfspId: config.dfspId,
+            tls: config.tls,
+            jwsSign: config.jwsSign,
+            jwsSigningKey: config.jwsSigningKey,
+            wso2Auth: config.wso2Auth
+        });
+
+        this._backendRequests = new BackendRequests({
+            logger: this._logger,
+            backendEndpoint: config.backendEndpoint,
+            dfspId: config.dfspId
+        });
+
+        this._checkIlp = config.checkIlp;
+
+        this._ilp = new Ilp({
+            secret: config.ilpSecret
+        });
+    }
+
+    /**
+     * Queries the backend API to get the authorization details and makes a 
+     * callback to the originator with the result
+     */
+    async postAuthorizations(authorizationsReq, sourceFspId) {
+        try {
+            //Converts a mojaloop authorizationsReq data to internal form
+            const internalForm = shared.mojaloopAuthorizationsReqToInternal(authorizationsReq);
+
+            // make a call to the backend to ask for a authorizations response
+            const response = await this._backendRequests.getSignedChallenge(internalForm);
+            if (!response) {
+                // make an error callback to the source fsp
+                return 'No response from backend';
```

Comment:
shouldn't be a `throw` instead `return` ?

Reply:
Followed existing pattern.